worker/caas/broker.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package caas
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+
+	"github.com/juju/juju/caas"
+	"github.com/juju/juju/environs"
+	"github.com/juju/juju/worker/catacomb"
+)
+
+var logger = loggo.GetLogger("juju.worker.caas")
+
+// ConfigObserver exposes a model configuration and a watch constructor
+// that allows clients to be informed of changes to the configuration.
+type ConfigObserver interface {
+	CloudSpec() (environs.CloudSpec, error)
+}
+
+// Config describes the dependencies of a Tracker.
+//
+// It's arguable that it should be called TrackerConfig, because of the heavy
+// use of model config in this package.
+type Config struct {
+	Observer      ConfigObserver
+	NewBrokerFunc caas.NewBrokerFunc
+}
+
+// Validate returns an error if the config cannot be used to start a Tracker.
+func (config Config) Validate() error {
+	if config.Observer == nil {
+		return errors.NotValidf("nil Observer")
+	}
+	if config.NewBrokerFunc == nil {
+		return errors.NotValidf("nil NewBrokerFunc")
+	}
+	return nil
+}
+
+// Tracker loads an environment, makes it available to clients, and updates
```

Comment:
s/environment/CAAS broker/
(please check for other copy pasta)

Reply:
done